EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -41,33 +42,52 @@ public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder,
                 var typeMapping = (RelationalTypeMapping?)property.FindTypeMapping()
                     ?? _typeMappingSource.FindMapping((IProperty)property);
                 var propertyType = Nullable.GetUnderlyingType(property.ClrType) ?? property.ClrType;
-                if (propertyType.IsEnum
-                    && typeMapping != null
-                    && !propertyType.IsDefined(typeof(FlagsAttribute), true)
-                    && property.GetColumnName(tableIdentifier) is { } columnName)
+                if (!propertyType.IsEnum
+                    || typeMapping == null
+                    || propertyType.IsDefined(typeof(FlagsAttribute), true)
+                    || property.GetColumnName(tableIdentifier) is not {} columnName)
                 {
-                    var enumValues = Enum.GetValues(propertyType);
-                    if (enumValues.Length <= 0)
-                    {
-                        continue;
-                    }
+                    continue;
+                }
+
+                var enumValues = Enum.GetValues(propertyType).Cast<object>().Select(typeMapping.GenerateSqlLiteral).ToList();
+                if (enumValues.Count == 0)
+                {
+                    continue;
+                }
 
-                    sql.Clear();
+                sql.Clear();
 
-                    sql.Append(_sqlGenerationHelper.DelimitIdentifier(columnName));
+                sql.Append(_sqlGenerationHelper.DelimitIdentifier(columnName));
+
+                // the list is a contiguous range if the following are true
+                //  first value is int x
+                //  last value is int y
+                //  y - x is one less than the number of total values
+                if (int.TryParse(enumValues[0], out var firstValue)
```

Comment:
Shouldn't we first sort the values? There's no guarantee that the order of values returned by Enum.GetValues above is sorted etc.

Reply:
You're absolutely right, they could be out of order. I'll get that updated and add a test for it also.